src/Rules/Register.hs

```diff
@@ -83,8 +83,11 @@ copyConf rs context@Context {..} conf = do
     depPkgIds <- fmap stdOutToPkgIds . askWithResources rs $
       target context (GhcPkg Dependencies stage) [pkgName package] []
     need =<< mapM (\pkgId -> packageDbPath stage <&> (-/- pkgId <.> "conf")) depPkgIds
-    buildWithResources rs $
-      target context (GhcPkg Clone stage) [pkgName package] [conf]
+    unlessM (doesFileExist conf) $ do
+      buildWithResources rs $
+        target context (GhcPkg Unregister stage) [pkgName package] []
```

Comment:
Why do we need to unregister the package if the file doesn't exist? How would the package end up being registered even if the file does not exist?

I would have thought we wanted:
```haskell
whenM (doesFileExist conf) {- unregister -}
{- register -}
```

Reply:
Looking at #543 I kind of see the reasoning for the unconditional `unregister` now.
That might warrant a comment though.

Reply:
> Why do we need to unregister the package if the file doesn't exist? How would the package end up being registered even if the file does not exist?

only if you deliberately deleted `[pkg-id].conf` file and then `register`. I think it is a rare case though...

Reply:
I'm also unsure if we just want to skip this, or rather always *reregister* the package. In case the `conf` file has changed, even though we likely want to assume that no-one really changes the bootstrap compiler while building right? In that case we can just always unregister+register.

Reply:
> just always unregister+register.

the problem is if so, when you run `./build.sh` after a full build, the `unregister+register`s will be repeated, which doesn't look sane to me.

Reply:
Can we have Shake somehow verify that the `.conf` file is identical?

Reply:
> Can we have Shake somehow verify that the .conf file is identical?

Do you mean the result of `describe`? I doubt if this kind of `Stdout` can be tracked...

Reply:
Alright. Let's just add a comment why we do the unreg+reg (e.g. the comment from #543) and what the justification for not re-registering (full build should not repeatedly unregister/register).